src/App.tsx

```diff
@@ -1,24 +1,204 @@
+  const [error, setError] = useState<string | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingIds, setLoadingIds] = useState<number[]>([0]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(data => setTodos(data));
+  }, []);
+
+  useEffect(() => {
+    let timerId:NodeJS.Timeout;
+
+    if (error) {
+      timerId = setTimeout(() => {
+        setError(null);
+      }, 2000);
+    }
+
+    return () => {
+      clearTimeout(timerId);
+    };
+  }, [error]);
+
+  const createTodo = useCallback(async (title: string) => {
+    const newTodo = {
+      title,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    setTempTodo({
+      id: 0,
+      title,
+      completed: false,
+      userId: USER_ID,
+    });
+
+    try {
+      const createdTodo = await addTodo(newTodo);
+
+      setTempTodo(null);
+
+      setTodos(prevTodos => [...prevTodos, createdTodo]);
+    } catch {
+      setError('Unable to create a todo');
+    }
+  }, []);
+
+  const removeTodo = useCallback(async (id: number) => {
+    setLoadingIds(prevIds => [...prevIds, id]);
+
+    try {
+      await deleteTodo(id);
+      setTodos(prevTodos => (
+        prevTodos.filter(todo => todo.id !== id)));
+
+      setLoadingIds([0]);
+    } catch {
+      setError('Unable to delete a todo');
+    }
+  }, []);
+
+  const updateStatus = useCallback(async (id: number, status: boolean) => {
+    setLoadingIds(prevIds => [...prevIds, id]);
+
+    try {
+      await updateTodo(id, { completed: status });
+      setTodos(prevTodos => prevTodos.map(todo => {
+        if (todo.id !== id) {
+          return todo;
+        }
+
+        return {
+          ...todo,
+          completed: status,
+        };
+      }));
+
+      setLoadingIds([0]);
+    } catch {
+      setError('Unable to update a todo');
+    }
+  }, []);
+
+  const handleSortType = (type: SortType) => {
+    setSortBy(type);
+  };
+
+  const handleTitleError = (message: string) => {
+    setError(message);
+  };
+
+  const handleDeleteAllCompleted = () => {
+    const completedTodos = todos.filter(todo => todo.completed);
+    const deletePromises = completedTodos.map(todo => removeTodo(todo.id));
+
+    Promise.all(deletePromises)
+      .then(() => (
+        setTodos(prevTodos => (
+          prevTodos.filter(todo => !todo.completed)))
+      ));
+  };
+
+  const editTodo = async (currentId: number, data: Partial<Todo>) => {
+    setLoadingIds(prevIds => [...prevIds, currentId]);
+
+    try {
+      const updatedTodo = await updateTodo(currentId, data);
+
+      setTodos(prevTodos => prevTodos.map(prevTodo => (
+        prevTodo.id === updatedTodo.id
+          ? updatedTodo
+          : prevTodo
+      )));
+
+      setLoadingIds([0]);
+    } catch {
+      setError('Unable to update a todo');
+    }
+  };
+
+  const filters: string[] = [
+    SortType.All,
+    SortType.Active,
+    SortType.Completed,
+  ];
```

Comment:
You don't have to create an array of filters here and pass it with props, instead use Object.values(Enum) directly in your footer